packages/contracts-bedrock/scripts/DeploySuperchain.s.sol

```diff
@@ -57,64 +58,64 @@ import { Solarray } from "scripts/libraries/Solarray.sol";
-        ProtocolVersion recommendedProtocolVersion;
-        ProtocolVersion requiredProtocolVersion;
-        Roles roles;
+    using stdToml for string;
+
+    // All inputs are set in storage individually. We put any roles first, followed by the remaining
+    // inputs. Inputs are internal and prefixed with an underscore, because we will expose a getter
+    // method that returns the input value. We use a getter method to allow us to make assertions on
+    // the input to ensure it's valid before returning it. We also intentionally do not use a struct
+    // to hold all inputs, because as features are developed the set of inputs will change, and
+    // modifying structs in Solidity is not very simple.
+
+    // Role inputs.
+    address internal _guardian;
+    address internal _protocolVersionsOwner;
+    address internal _proxyAdminOwner;
+
+    // Other inputs.
+    bool internal _paused;
+    ProtocolVersion internal _recommendedProtocolVersion;
+    ProtocolVersion internal _requiredProtocolVersion;
+
+    // These `set` methods let each input be set individually. The selector of an input's getter method
+    // is used to determine which field to set.
+    function set(bytes4 _sel, address _address) public {
+        require(_address != address(0), "DeploySuperchainInput: cannot set null address");
+        if (_sel == this.guardian.selector) _guardian = _address;
+        else if (_sel == this.protocolVersionsOwner.selector) _protocolVersionsOwner = _address;
+        else if (_sel == this.proxyAdminOwner.selector) _proxyAdminOwner = _address;
+        else revert("DeploySuperchainInput: unknown selector");
     }
 
-    struct Roles {
-        address guardian;
-        address protocolVersionsOwner;
-        address proxyAdminOwner;
+    function set(bytes4 _sel, bool _value) public {
+        if (_sel == this.paused.selector) _paused = _value;
+        else revert("DeploySuperchainInput: unknown selector");
     }
 
-    // This flag tells us if all inputs have been set. An `input()` getter method that returns all
-    // inputs reverts if this flag is false. This ensures the deploy script cannot proceed until all
-    // inputs are validated and set.
-    bool public inputSet = false;
-
-    // The full input struct is kept in storage. It is not exposed because the return type would be
-    // a tuple, and it's more convenient for the return type to be the struct itself. Therefore the
-    // struct is exposed via the `input()` getter method below.
-    Input internal inputs;
+    function set(bytes4 _sel, ProtocolVersion _value) public {
+        require(ProtocolVersion.unwrap(_value) != 0, "DeploySuperchainInput: cannot set null protocol version");
+        if (_sel == this.recommendedProtocolVersion.selector) _recommendedProtocolVersion = _value;
+        else if (_sel == this.requiredProtocolVersion.selector) _requiredProtocolVersion = _value;
+        else revert("DeploySuperchainInput: unknown selector");
+    }
```

Comment:
This feels error prone. 

Is there anything here or in the tests that ensures that all config values have a corresponding setter path?

Reply:
Error prone in the sense that we might forget a setter? That's fair. How about if we use the setter methods in `loadInputFile`? We currently just write to storage slots directly, but using the setter seems preferable because then any safety checks are also run (i.e. don't allow setting null values).

This way, by testing `loadInputFile` (which we do in `test_loadInputFile_succeeds`) we get validation that all setter methods are present, for free

Reply:
Pushed bd36605 which uses setters